forte/processors/data_augment/algorithms/UDA.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file implements the UDA(Unsupervised Data Augmentation)
+algorithm as described in
+Unsupervised Data Augmentation for Consistency Training
+(https://arxiv.org/abs/1904.12848)
+(https://github.com/google-research/uda)
+"""
+import tensorflow as tf
+
+def kl_divergence_log_probs(log_p, log_q):
+    r"""
+    This function calculates the Kullback-Leibler Divergence
+    between two distributions. The inputs are batched log probabilities.
+
+    Args:
+        - log_p: A batch of probabilities with a shape of [batch_size, dimension].
+        - log_q: A batch of probabilities with a shape of [batch_size, dimension].
+    Returns:
+        The KL divergence for the probability pairs with a shape of [batch_size]
+    """
+    p = tf.exp(log_p)
```

Comment:
This operation is probably not robust (e.g., numeric overflow/underflow). 
- Is there any robust open-source implementation we can refer to?
- Or we may note this explicitly in the doc